stm/stm.ml

```diff
@@ -370,6 +321,12 @@ end = struct (* {{{ *)
   open Printf
 
   let print_dag vcs () =
+
+    (* Due to threading, this will be called from the toplevel with
+       we_are_parsing set to true, as indeed, the toplevel is waiting
+       for input . What a race! XD *)
+    Flags.we_are_parsing := false;
```

Comment:
Anyway, Dag.print is for debugging only... but flipping the boolean is wrong.  What you want to do is to catch the exception and stop printing in that case.  Let's call it "best effort debug print"